velox/core/PlanNode.h

```diff
@@ -1768,6 +1768,74 @@ class MergeJoinNode : public AbstractJoinNode {
   static PlanNodePtr create(const folly::dynamic& obj, void* context);
 };
 
+struct IndexJoinCondition;
+using IndexJoinConditionPtr = std::shared_ptr<IndexJoinCondition>;
+struct IndexJoinCondition : public ISerializable {
+  /// References to an index table column.
+  FieldAccessTypedExprPtr key;
+
+  IndexJoinCondition(FieldAccessTypedExprPtr _key) : key(std::move(_key)) {}
+
+  folly::dynamic serialize() const override;
+
+  virtual std::string toString() const = 0;
+
+  static IndexJoinConditionPtr create(const folly::dynamic& obj, void* context);
```

Comment:
Should we remove this from the base class?

Reply:
Removed.